src/rpc/mining.cpp

```diff
@@ -132,7 +128,10 @@ static bool GenerateBlock(ChainstateManager& chainman, CBlock& block, uint64_t&
         throw JSONRPCError(RPC_INTERNAL_ERROR, "ProcessNewBlock, block not accepted");
     }
 
-    block_hash = block.GetHash();
+    if (block.GetHash().IsNull()) {
+        throw JSONRPCError(RPC_MISC_ERROR, "Failed to make block.");
+    }
```

Comment:
I tend to agree with @theStack that throwing here is a change of failure modes for `generateBlocks`, impacting `generatetodescriptor` and `generatetoaddress`.

However, I'm not clear on when this condition occurs. `CBlock::GetHash()` resolves to `SerializeHash(*this)`, and I didn't see a failure possible in the serialization logic (through a cursory glance).

Options seem to be:
* retain existing behavior by returning false in this case
* fail more noisily, given that this is not an expected error, as here

Reply:
It's a very strange and unintuitive interface. The passed out hash will be null if `block.nNonce == std::numeric_limits<uint32_t>::max()`. The nullness of that out-param is what is used to determine whether the outer while loop should increment the height or try again at the same height.